moveit_kinematics/lma_kinematics_plugin/src/lma_kinematics_plugin.cpp

```diff
@@ -88,68 +88,66 @@ bool LMAKinematicsPlugin::initialize(const moveit::core::RobotModel& robot_model
   {
-    ROS_ERROR_NAMED("lma", "Group '%s' includes joints that have more than 1 DOF", group_name.c_str());
+    RCLCPP_ERROR(node_->get_logger(), "Group '%s' includes joints that have more than 1 DOF", group_name.c_str());
     return false;
   }
 
   KDL::Tree kdl_tree;
 
   if (!kdl_parser::treeFromUrdfModel(*robot_model.getURDF(), kdl_tree))
   {
-    ROS_ERROR_NAMED("lma", "Could not initialize tree object");
+    RCLCPP_ERROR(node_->get_logger(), "Could not initialize tree object");
     return false;
   }
   if (!kdl_tree.getChain(base_frame_, getTipFrame(), kdl_chain_))
   {
-    ROS_ERROR_NAMED("lma", "Could not initialize chain object");
+    RCLCPP_ERROR(node_->get_logger(), "Could not initialize chain object");
     return false;
   }
 
   for (const robot_model::JointModel* jm : joint_model_group_->getJointModels())
   {
     if (jm->getType() == moveit::core::JointModel::REVOLUTE || jm->getType() == moveit::core::JointModel::PRISMATIC)
     {
-      ik_chain_info_.joint_names.push_back(joint_model_group->getJointModelNames()[i]);
-      const std::vector<moveit_msgs::msg::JointLimits>& jvec =
-          joint_model_group->getJointModels()[i]->getVariableBoundsMsg();
-      ik_chain_info_.limits.insert(ik_chain_info_.limits.end(), jvec.begin(), jvec.end());
+      joints_.push_back(jm);
+      joint_names_.push_back(jm->getName());
     }
   }
   dimension_ = joints_.size();
 
   // Get Solver Parameters
-  lookupParam("max_solver_iterations", max_solver_iterations_, 500);
-  lookupParam("epsilon", epsilon_, 1e-5);
-  lookupParam("orientation_vs_position", orientation_vs_position_weight_, 0.01);
+  lookupParam(node_,"max_solver_iterations", max_solver_iterations_, 500);
+  lookupParam(node_,"epsilon", epsilon_, 1e-5);
+  lookupParam(node_,"orientation_vs_position", orientation_vs_position_weight_, 0.01);
 
   bool position_ik;
-  lookupParam("position_only_ik", position_ik, false);
+  lookupParam(node_,"position_only_ik", position_ik, false);
   if (position_ik)  // position_only_ik overrules orientation_vs_position
     orientation_vs_position_weight_ = 0.0;
   if (orientation_vs_position_weight_ == 0.0)
-    ROS_INFO_NAMED("lma", "Using position only ik");
+    RCLCPP_INFO(node_->get_logger(), "Using position only ik");
 
   // Setup the joint state groups that we need
   state_.reset(new robot_state::RobotState(robot_model_));
 
   fk_solver_.reset(new KDL::ChainFkSolverPos_recursive(kdl_chain_));
 
   initialized_ = true;
-  ROS_DEBUG_NAMED("lma", "LMA solver initialized");
+  RCLCPP_DEBUG(node_->get_logger(), "LMA solver initialized");
   return true;
 }
 
-bool LMAKinematicsPlugin::timedOut(const ros::WallTime& start_time, double duration) const
+bool LMAKinematicsPlugin::timedOut(const std::chrono::system_clock::time_point& start_time, double duration) const
 {
-  return ((ros::WallTime::now() - start_time).toSec() >= duration);
+  return ((std::chrono::system_clock::now() - start_time).count() >= duration);
```

Comment:
which is the units of duration?

Reply:
https://github.com/AcutronicRobotics/moveit2/pull/75/commits/c2f4ba6a8fee1de985726e4d0d6156ce9704db36